tracecontext.go

```diff
@@ -169,23 +169,61 @@ type TraceState struct {
 
 // NewTraceState returns a TraceState based on entries.
 func NewTraceState(entries ...TraceStateEntry) TraceState {
-	out := TraceState{}
+	var out TraceState
 	var last *TraceStateEntry
+	var modified []TraceStateEntry
+	recorded := make(map[string]uint8)
```

Comment:
This could also be named `seen` or `observed`, I think it works either way but I'm happy to rename it.